shared/teams/edit-team-welcome-message/index.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import * as TeamsGen from '../../actions/teams-gen'
+import * as Container from '../../util/container'
+import * as Constants from '../../constants/teams'
+import * as Types from '../../constants/types/teams'
+import {computeWelcomeMessageText} from '../../chat/conversation/messages/cards/team-journey/util'
+
+type Props = Container.RouteProps<{teamID: Types.TeamID}>
+
+// welcomeMessageMaxLen is duplicated at
+// go/chat/server.go:welcomeMessageMaxLen; keep the values in sync!
+const welcomeMessageMaxLen = 400
+
+const EditTeamWelcomeMessage = (props: Props) => {
+  const teamID = Container.getRouteProps(props, 'teamID', Types.noTeamID)
+
+  if (teamID === Types.noTeamID) {
+    throw new Error(`There was a problem loading the welcome message page, please report this error.`)
+  }
+
+  const waitingKey = Container.useSelector(_ => Constants.setWelcomeMessageWaitingKey(teamID))
+  const waiting = Container.useAnyWaiting(waitingKey)
+  const error = Container.useSelector(state => state.teams.errorInEditWelcomeMessage)
+  const origWelcomeMessage = Container.useSelector(state => {
+    const message = Constants.getTeamWelcomeMessageByID(state, teamID)
+    if (message) {
+      return message
+    }
+    throw new Error(`There was a problem loading the welcome message, please report this error.`)
```

Comment:
What would cause us to not load this? IMO this is extra code for a case that seems pretty unlikely and would be noticeable anyway (by the welcome message not populating).

Reply:
I don't think that's possible, because you have to be on the settings tab and the Edit button won't load until it's populated. Originally I had `message!` but wasn't sure if that's safe to do as far as defensive coding goes.

Reply:
If you don't think it's possible, I think that `message!` seems perfectly acceptable to me.